### PR TITLE
[vcpkg] Add x64-windows-static-md to pipeline

### LINKED ISSUE #15122: Elevate x64-windows-static-md to be an official triplet
While i have nothing against the triplet x64-windows-static , it does have a pitfall of assuming that developers want the CRT to be statically linked in the binary as well which is rarely the case.

Most developers or projects while building on Windows face the issues around DLLs with regards to getting them to load from the right locations while developing or package them along with their binaries, given that DLLs dont really contain an rpath.
To circumvent such frustrations a lot of developers choose to use static-md so that all libraries are statically linked.
There may be a lot of other reasons to choose static linking too but in neither of the cases linking against a static CRT is a good idea.

Most developer if given the option would probably choose static-md rather than static

Coupled this with the fact that i've seen static-md break far too often. 
I would propose elevating x64-windows-static-md to be an official triplet. 
I would even go so far as to even consider replacing static with static-md but thats if we're given a choice between static and static-md.

The alternative (static-without-md) is simply a bloat for the binary. In most cases its not something the developer really intends anyway.

### PR DESCRIPTION
**Pull request by ankurvdev**
Add x64-windows-static-md to CI Pipeline checks

- What does your PR fix? Fixes #15122 

- Which triplets are supported/not supported? Have you updated the CI baseline?

- Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?

### CODE REVIEW

**Comment by ankurverma85** [comment]
@JackBoosY, @BillyONeal

Looks like the ci pipeline run timed out after getting stuck
I'm guessing some transient infrastructure issues

Can someone with enough permissions restart the run or "re-run failed jobs"

**Comment by ankurverma85** [comment]
Looks like the checks all pass :) 
Hate to rush , but I'd love to merge this ASAP if there's no concerns/feedback

Have a few other pull requests that relate to x64-windows-static-md that I'd like to queue up after this and also possibly avoid a re-run/merge from other PRs

**Review by JackBoosY — APPROVED** [review]

**Comment by JackBoosY** [comment]
LGTM, ping for @BillyONeal review this PR.

**Review by BillyONeal — APPROVED** [review]

File: scripts/ci.baseline.txt [hunk 1]
```diff
@@ -58,6 +59,10 @@ caf:x64-uwp=fail
 caffe2:x86-windows=fail
 caffe2:arm64-windows=fail
 cairomm:x64-linux=fail
+# capnproto doesn't support arm, arm64, or UWP
```
Comment (BillyONeal) [inline scripts/ci.baseline.txt:62]: This looks like a broken merge.

File: scripts/ci.baseline.txt [hunk 2]
```diff
@@ -1635,8 +1773,44 @@ protozero:x64-uwp=fail
...
 clapack:x64-windows-static=skip
+clapack:x64-windows-static-md=skip
 clapack:x64-windows=skip
 clapack:x86-windows=skip
 clapack:x64-uwp=skip
 lapack-reference:arm64-windows=skip
 lapack-reference:arm-uwp=skip
+
+
+# failures for x64-windows-static-md
```
Comment (BillyONeal) [inline scripts/ci.baseline.txt:1784]: You should merge these into the sorted list.

**Comment by ankurverma85** [comment]
@BillyONeal, can we get this merged ?

**Comment by BillyONeal** [comment]
> @BillyONeal, can we get this merged ?

It LGTM but the current "community reviewer" is @vicroms so I'd like him to look at it.

**Pull request merged**